### ISSUE TITLE
Change Safe Combination overlaying checkbox

### ISSUE THREAD

**Issue by igor2x** [pwsafe/pwsafe#1081]
On Ubuntu 23.10 I have installed (bellow image):
1. deb package v1.18.2
2. flatpak package v1.18.2 [commited today to flathub](https://github.com/ronys/flathub/blob/pwsafe/org.pwsafe.pwsafe.yml#L2)
3. flatpak package v1.18.2 updated to the latest libraries versions

![safe_combination](https://github.com/pwsafe/pwsafe/assets/10895030/4e24e8a9-b88e-410a-ba0b-6e7d3db15d96)

Just wondering, is there any chance deb and flatpak are not made from the same source code?

---

Comparison between point 2 (right site of bellow image) and point 3 (left site of image).
![image](https://github.com/pwsafe/pwsafe/assets/10895030/95cea21f-9f58-4a1c-935b-f675e459bccd)

**Comment by igor2x** [comment]
I did a few tests... it looks this "missing" checkbox is deb package problem.

**DEB PACKAGE VERSION**
1. Open Password Safe and make sure when typing in safe combination password, that "Open as read only" in unchecked.
2. Manage | Change Safe Combination and there is no "Show Combination" checkbox.

![image](https://github.com/pwsafe/pwsafe/assets/10895030/0680eb36-946d-4ba2-bb62-7c61cf33ecac)

3. File | Change to R-O
4. File | Change to R-W and type in password.
5. Manage | Change Safe Combination and "Show Combination" checkbox appears.
![image](https://github.com/pwsafe/pwsafe/assets/10895030/8c2bf76a-1a88-4d15-be2e-e5208f496225)

**COMPARE TO FLATPAK VERSION**
After step 2 and step 5 in both cases there is "Show Combination" checkbox.
![image](https://github.com/pwsafe/pwsafe/assets/10895030/c155dd9e-efa5-4b46-b79e-58c89b696f0f)

It looks to me that missing checkbox is actually deb package problem and flatpak package works fine.

**Comment by igor2x** [comment]
I checked the dialogs in Password Safe 3.65 on Windows and I see there is checkbox "Show Master Password".
I assume then, the checkbox should always be on, so some issue in deb package.

![image](https://github.com/pwsafe/pwsafe/assets/10895030/c876d04b-ee96-485b-8904-bb4bf845b448)

**Comment by chrisawi** [comment]
There's some kind of layout bug here, possibly in wxWidgets. Running the Flathub version on Fedora 39 (using the default Adwaita GTK3 theme), I see no checkbox and the following output:
```
(pwsafe:2): Gtk-WARNING **: 13:41:27.295: Negative content height -2 (allocation 2, extents 2x2) while allocating gadget (node checkbutton, owner GtkCheckButton)

(pwsafe:2): Gtk-WARNING **: 13:41:27.295: for_size smaller than min-size (0 < 14) while measuring gadget (node check, owner GtkCheckButton)

(pwsafe:2): Gtk-WARNING **: 13:41:27.295: Negative content height -2 (allocation 0, extents 1x1) while allocating gadget (node check, owner GtkCheckButton)
```

The checkbox becomes visible after changing the theme to win32 in the GTK inspector.

**Comment by igor2x** [comment]
On my local computer:
1. Installed Password Safe v1.18.2 from flatpak/Flathub on Ubuntu 23.10 (top left on picture) and there is problem with checkbox postion.
2. Installed Pasword Safe v1.18.2 from flatpak/Flathub on Fedora 39 (bottom left on picture) and there is problem checkbox does not appear at all.
3. Locally updated wxwidgets to v3.2.4 version, locally build and installed on Ubuntu 23.10 (top right) and problem is fixed.
4. Locally updated wxwidgets to v3.2.4 version, locally build and installed on Fedora 39 (bottom right) and problem is fixed.

The result of above steps:
![image](https://github.com/pwsafe/pwsafe/assets/10895030/85945035-7c23-488e-b413-04f8abe3edd7)

@ronys all what is required is:
1. Open file [org.pwsafe.pwsafe.yml]( https://github.com/flathub/org.pwsafe.pwsafe/blob/master/org.pwsafe.pwsafe.yml) file.
2. Search and replace.
In line 38:
- current: tag: v3.2.2.1
- replace: tag: v3.2.4

In line 39:
- current: commit: a812fffda3fe686c94e24bff27e8effd96e4de64
- replace: commit: 085a136dcb11aca5b1102193f006f8056d5f0876

Note: New "tags" and "commit" numbers I have gotten from: https://github.com/wxWidgets/wxWidgets/tags

3. Rebuild flatpak. Probably now you can commit directly to flathub's org.pwsafe.pwsafe repository to master branch and if I read documentation correctly then you can publish new flatpak directly or just wait for 3 hours and it gets automatically published.

**Comment by igor2x** [comment]
Now I installed deb package v1.18.2 on Ubunu 23.10 again (after uninstalling it like a week ago) and checked the menu **Manage | Change Safe Combination** and problem is no longer there, like it was two weeks back with missing "Show Combination" check box. Interesting, maybe there was update in wxWidgets Ubuntu packet.

I am sure this is now deb package and not flatpak, because I have double checked and started program from /usr/bin/pwsafe and in Help | About there is no "fp4" that is tag I can see with flatpak package.

Bottom line, now the only issue is with flatpak. If code provided by @cassidyjames is merged and push to flatpak Flathub repo, and then this issue can be closed down as solved.

**Comment by igor2x** [comment]
Today I have tested the latest updated version of flatpak from Flathub and problem is solved.
Thanks for fixing the issue.
